stock_picking_group_by_partner_by_carrier/models/stock_picking.py

```diff
@@ -59,6 +59,8 @@ def action_cancel(self):
             return super().action_cancel()
 
     def _create_backorder(self):
+        if self.partner_id.disable_picking_grouping:
```

Comment:
:fire: this override is wrong. You should loop on self.
Found out because our CI is failing :wink:

Reply:
also, the check below for setting the ctx key should be probably done by grouping per picking type? :thinking:

Reply:
What is strange is this issue wasn't triggered before on the existing check, we were lucky.
I reimplemented the override to loop on each picking, your review is welcome.